src/claimtrie.cpp

```diff
@@ -137,6 +156,109 @@ void CClaimTrieNode::reorderClaims(supportMapEntryType& supports)
     std::make_heap(claims.begin(), claims.end());
 }
 
+// trieCache is an input parameter on enable; an output parameter on disable.
+CClaimTrie* CClaimTrie::enableNormalizationFork(bool enable, int nHeight, CClaimTrieCache& trieCache)
+{
+    LogPrintf("%s: Normalize Name Fork is %s\n", __func__, (enable ? "ENABLED" : "DISABLED"));
+
+    if (enable) {
+        if (fNormalize) {
+            return this;
+        }
+
+        std::vector<namedNodeType> nodes = flattenTrie();
+        LogPrintf("Pre-normalization fork claim trie nodes = %lu\n", nodes.size());
+
+        if (normalizationForkStateInstance.nonNormalizedTrie == NULL) {
+            normalizationForkStateInstance.nonNormalizedTrie = this;
+        }
+
+        if (normalizationForkStateInstance.nonNormalizedTrieCache == NULL) {
+            normalizationForkStateInstance.nonNormalizedTrieCache = boost::make_shared<CClaimTrieCache>(this);
+            normalizationForkStateInstance.nonNormalizedTrieCache->cloneState(trieCache);
+            LogPrintf("%s: Cloned Merkle hash %s, Original %s\n", __func__, normalizationForkStateInstance.nonNormalizedTrieCache->getMerkleHash(true).GetHex().c_str(), trieCache.getMerkleHash(true).GetHex().c_str());
```

Comment:
This line is brutally expensive.

Reply:
In what way?  Show me the measurements.

Reply:
We presently have 5.5million nodes in the trie. `getMerkleHash(true)` triggers a rehash on every one of those. I just measured the call at 3 microseconds each. This was on a non-optimized build, but even if you figure 1 microsecond per hash call you're looking at 10 seconds of CPU hit for that one line.